File: SelectionSort.cpp
#include <iostream>
#include <vector>
#include <climits>

using namespace std;

int main(){

    int minIndex=0;
    int arr[] = {3,2,1,5,6};
    int n = 6;
    
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
    
    cout<<endl;
    
    for(int i=0; i<n-1; i++){
        minIndex = i;
        
        for(int j=i+1; j<n; j++){
            if(arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        swap(arr[i], arr[minIndex]);
    }
    
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
    
    
    return 0;
}

File: SelectionSort.java
public class Main {

    static void selectionSort(int[] arr, int n) {

        for (int i = 0; i < n - 1; i++) {

            int minIndex = i;

            for (int j = i + 1; j < n; j++) {

                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            // Swap arr[i] and arr[minIndex]
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }

    public static void main(String[] args) {

        int[] arr = {64, 25, 12, 22, 11};

        int n = arr.length;

        selectionSort(arr, n);

        System.out.println("Sorted array:");

        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
